assets/asset_resolver.h

```diff
@@ -21,6 +21,17 @@ class AssetResolver {
 
   virtual bool IsValid() const = 0;
 
+  //----------------------------------------------------------------------------
+  /// @brief      Certain asset resolvers need to be preserved when the asset
+  ///             directory is updated or the isolate recreated. While some
+  ///             could be recated from the settings object, the Android
+  ///             specific asset resolvers require a reference to the JNI to
+  ///             create.
+  ///
+  /// @return     Returns whether this resolver should be preserved.
+  ///
+  virtual bool ShouldPreserve() const = 0;
```

Comment:
Can we come up with a better name for this?

Maybe something like `IsValidAfterIsolateRestart`?

"Should" is a bit of a weak word to use (we usually prefer "must"), and it'd be helpful to have more context when reading the code about when or why I should care about this parameter.

Reply:
How about something like:

Asset resolvers that are valid after a reload or a restart must be preserved and inserted into the newly created asset manager or run configuration. This allows the tooling to avoid copying assets through the device devFS on start, which has
reliability issues on iOS and is slow when the asset size is larger.

Reply:
That might help with the docs, but I'm looking for a better member name than `ShouldPreserve`. Does `IsValidAfterIsolateRestart`?

And it this something we could combine with `IsValid` somehow if we parameterized it? Perhaps with a default value like `IsValid(bool isolate_restarted = false)`?

Reply:
Its not just restarts though, its also the first hot reload. `IsValidAfterAssetManagerChange` maybe?

Reply:
Sure. Just something that gives me a slightly better clue as I go to use the API and read the code. And then in the docs we can describe when an asset manager changes or why I'd want to check this.